src/main/java/com/google/devtools/build/lib/actions/ExecutionRequirements.java

```diff
@@ -105,6 +105,96 @@ public String parseIfMatches(String tag) throws ValidationException {
     }
   }
 
+  /** An execution requirement that is split into a key and a parseable value. */
+  @AutoValue
+  public abstract static class TaggedRequirement<ValueType> {
+    /**
+     * Thrown when a the {@link #validator()} method returns an error.
+     */
+    public static class ValidationException extends Exception {
+      private final String tagValue;
+
+      /**
+       * Creates a new {@link ValidationException}.
+       *
+       * @param tagValue the erroneous value that was parsed from the tag.
+       * @param errorMsg an error message that tells the user what's wrong with the value.
+       */
+      public ValidationException(String tagValue, String errorMsg) {
+        super(errorMsg);
+        this.tagValue = tagValue;
```

Comment:
Shouldn't we guarantee in some way that the error message contains the tag value?

Reply:
Good question. I can go in either direction, but I'd like to point out that this is copied/pasted from `ParseableRequirement.ValidationException`. Similarly, the error handler in `SkylarkActionFactory.java` is directly influenced by the one from `TestTargetProperties.java`.

What's the dominant convention? (Throwers of `InvalidPackageNameException` don't include bad values in their error message, while throwers of `EvalException` sometimes do.)